protocol/messenger_communities.go

```diff
@@ -4097,21 +4120,14 @@ func (m *Messenger) CheckAllCommunityChannelsPermissions(request *requests.Check
 		return nil, err
 	}
 
-	var addresses []gethcommon.Address
-
-	if len(request.Addresses) == 0 {
-		accounts, err := m.settings.GetActiveAccounts()
-		if err != nil {
-			return nil, err
-		}
+	sharedAddresses, err := m.GetRevealedAccounts(request.CommunityID, common.PubkeyToHex(&m.identity.PublicKey))
```

Comment:
Both function do the `GetRevealedAccounts` call, so it can be moved to the `checkCommunityPermissions` function (or whatever new name)